src/SearchParameters/index.js

```diff
@@ -93,6 +95,10 @@ function findArray(array, searchedValue) {
 function SearchParameters(newParameters) {
   var params = newParameters ? SearchParameters._parseNumbers(newParameters) : {};
 
+  if (params.userToken !== undefined && !isValidUserToken(params.userToken)) {
+    delete params.userToken;
+    logger.warn('The `userToken` parameter is invalid. Format: [a-zA-Z0-9_-]{1,64}');
```

Comment:
Since it is just a warning and it can have no apparent impact on the app behavior at first, I would add more information to the error message to precise what are the impact of having an invalid `userToken`, with a link to the documentation if possible.